Documentation/buildtools-repo-compose.md

```diff
@@ -0,0 +1,83 @@
+# Implementing [Repo API](RepoCompose.md) compose commands in buildtools
+
+## Why not RepoUtil?
+
+[RepoUtil][RepoUtil] derives `consumes` and `change` behavior from the contents of project.json files in a project, considering them the *source of truth* for dependency information. In some repositories, dependency information is solely stored in project.json files. When adding Repo API support in those repositories, it makes sense to use RepoUtil as-is.
+
+However, repositories that currently use buildtools tend to store some dependency information in other files, for example package versions used in MSBuild project files. There is already a *source of truth* independent of project.jsons in these repositories: [dependencies.props](https://github.com/dotnet/corefx/blob/713e104c72538f5b42c75882a337178ad8e4229b/dependencies.props). This doc assumes that buildtools-using repositories will want to continue using a non-project.json source of truth. Repositories that use buildtools are corefx, coreclr, wcf, and buildtools. (The current plan is to also convert core-setup to use buildtools, but that project doesn't have a dependencies.props source of truth as of writing. See [core-setup #283](https://github.com/dotnet/core-setup/issues/283).)
+
+This doc proposes the way that Repo API Compose commands will be implemented in BuildTools, with a non-project.json source of truth.
+
+
+## Overview
+
+The way `consumes`, `change`, and `verify` interact make it so they can be simplified by checking the `consumes` output into the repo as "consumes.json" and using it as the source of truth.
+
+ * `consumes` reads the contents of consumes.json and writes it out verbatim.
+
+ * `change` takes a modified consumes.json and overwrites the checked-in consumes.json file. Then, it changes the project contents so they match the new source of truth.
+
+ * `verify` reads consumes.json and ensures its contents are self-consistent and match the repository contents. This allows devs to check if their project.json or other modifications are valid.
+   * This is not a documented Repo API Compose command, but [RepoUtil][RepoUtil] offers it and the existing buildtools target `/t:VerifyDependencies` is an analogue.
+
+The `produces` command scans artifacts of a completed build for files that the repository produced and intends to publish. More detailed documentation will be available in BuildToolsRepoApiProducesImplementation.md [~~link~~](BuildToolsRepoApiProducesImplementation.md).
+
+The Repo API Compose command document [describes the workflow to update a version](RepoCompose.md#change), but to go into more detail, the corresponding flow in the Buildtools implementation is:
+
+ 1. `run consumes > temp.json`
```

Comment:
Seems to me like this round-trip is just useless and instead we should just have a convention of a checked in consumes.json file that people can directly get and/or edit it.